hedera-node/src/main/java/com/hedera/services/queries/token/GetAccountNftInfosAnswer.java

```diff
@@ -0,0 +1,182 @@
+import com.hederahashgraph.api.proto.java.TokenGetAccountNftInfosResponse;
+import com.hederahashgraph.api.proto.java.TokenNftInfo;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.hedera.services.utils.SignedTxnAccessor.uncheckedFrom;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_ACCOUNT_ID;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.QUERY_RANGE_LIMIT_EXCEEDED;
+import static com.hederahashgraph.api.proto.java.ResponseType.COST_ANSWER;
+
+public class GetAccountNftInfosAnswer implements AnswerService {
+    public static final String ACCOUNT_NFT_INFO_CTX_KEY = GetAccountNftInfosAnswer.class.getSimpleName() + "_accountNftInfos";
+    private OptionValidator validator;
+
+    public GetAccountNftInfosAnswer(OptionValidator validator) {
+        this.validator = validator;
+    }
+
+    @Override
+    public boolean needsAnswerOnlyCost(Query query) {
+        return COST_ANSWER == query.getTokenGetAccountNftInfos().getHeader().getResponseType();
+    }
+
+    @Override
+    public boolean requiresNodePayment(Query query) {
+        return typicallyRequiresNodePayment(query.getTokenGetAccountNftInfos().getHeader().getResponseType());
+    }
+
+    @Override
+    public Response responseGiven(Query query, StateView view, ResponseCodeEnum validity, long cost) {
+        return responseFor(query, view, validity, cost, NO_QUERY_CTX);
+    }
+
+    @Override
+    public Response responseGiven(
+            Query query,
+            StateView view,
+            ResponseCodeEnum validity,
+            long cost,
+            Map<String, Object> queryCtx
+    ) {
+        return responseFor(query, view, validity, cost, Optional.of(queryCtx));
+    }
+
+    @Override
+    public ResponseCodeEnum checkValidity(Query query, StateView view) {
+        var accountNftInfoQuery = query.getTokenGetAccountNftInfos();
+        AccountID id = accountNftInfoQuery.getAccountID();
+
+        if (accountNftInfoQuery.getStart() > accountNftInfoQuery.getEnd()) {
+            // TODO: Introduce another / Modify current query range error
+            return QUERY_RANGE_LIMIT_EXCEEDED;
+        }
+
+        var validity = validator.nftMaxQueryRangeCheck(accountNftInfoQuery.getStart(), accountNftInfoQuery.getEnd());
+        if (validity != OK) {
+            return validity;
+        }
+
+        validity = validator.queryableAccountStatus(id, view.accounts());
+        if (validity != OK) {
+            return validity;
+        }
+
+        var nftCount = view.accountNftsCount(id);
+        if (!(
+                accountNftInfoQuery.getStart() >= 0 &&
+                accountNftInfoQuery.getEnd() >= 0 &&
+                accountNftInfoQuery.getStart() < nftCount &&
+                accountNftInfoQuery.getEnd() < nftCount
+        )) {
```

Comment:
I think that we can write the check in a more understanding way for example switching the logic reverse (removing `!`). We have to be very clear on boundaries f.e
- can `start = end`? 
- is `end` inclusive?

Reply:
1. Yes, because the range is [start, end)
2. No

Reply:
I think here this is enough since we have more checks for interval before that:

`if (!(accountNftInfoQuery.getStart() >= 0 && accountNftInfoQuery.getEnd() < nftsCount)) {
			return QUERY_RANGE_LIMIT_EXCEEDED;
		}`